src/main/java/uk/gov/pay/adminusers/service/GovUkPayAgreementService.java

```diff
@@ -0,0 +1,35 @@
+package uk.gov.pay.adminusers.service;
+
+import com.google.inject.Inject;
+import org.slf4j.Logger;
+import uk.gov.pay.adminusers.logger.PayLoggerFactory;
+import uk.gov.pay.adminusers.model.GovUkPayAgreement;
+import uk.gov.pay.adminusers.persistence.dao.GovUkPayAgreementDao;
+import uk.gov.pay.adminusers.persistence.entity.GovUkPayAgreementEntity;
+
+import java.time.LocalDateTime;
+import java.util.Optional;
+
+import static java.lang.String.format;
+
+public class GovUkPayAgreementService {
+
+    private static Logger logger = PayLoggerFactory.getLogger(GovUkPayAgreementService.class);
+    private final GovUkPayAgreementDao agreementDao;
+    
+    @Inject
+    public GovUkPayAgreementService(GovUkPayAgreementDao govUkPayAgreementDao) {
+        this.agreementDao = govUkPayAgreementDao;
+    }
+
+
+    public Optional<GovUkPayAgreement> findGovUkPayAgreementByServiceId(int serviceId) {
+        return agreementDao.findByServiceId(serviceId)
+                .map((GovUkPayAgreementEntity::toGovUkPayAgreement));
+    }
+
+    public void doCreate(int serviceId, String email, LocalDateTime agreementTime) {
+        logger.info(format("Creating govuk pay agreement for service %s", serviceId));
```

Comment:
Really minor, but can we be branding-compliant and have this as “GOV.UK Pay agreement” please? 😛

Reply:
👍